wicket-core/src/main/java/org/apache/wicket/page/PageAccessSynchronizer.java

```diff
@@ -93,8 +94,8 @@ public Duration getTimeout(int pageId)
 	 */
 	public void lockPage(int pageId) throws CouldNotLockPageException
 	{
-		final Thread thread = Thread.currentThread();
-		final PageLock lock = new PageLock(pageId, thread);
+		final RequestCycle cycle = RequestCycle.get();

```

Comment:
Nothing from me - I didn't have any use for this yet, so I'm not sure how useful this is for components and how it should work exactly.